docs/architecture-overview.md

```diff
@@ -0,0 +1,159 @@
+## Architecture Overview
+
+### Unidirectional data flow
+
+Firefox Preview's presentation layer architecture is based on the concept of "unidirectional data flow." This is a popular approach in client side development, especially on the web, and is core to Redux, MVI, Elm Architecture, and Flux.  Our architecture is not identical to any of these (and they are not identical to each other), but the base concepts are the same. For a basic understanding of the motivations and approach, see [the official Redux docs](https://redux.js.org/basics/data-flow).  For an article on when unidirectional data flow is and is not a good approach, see [this](https://medium.com/swlh/the-case-for-flux-379b7d1982c6). These are both written from the perspective of React.js developers, but the concepts are largely the same.
+
+Our largest deviation from these architectures is that while they each recommend one large, global store of data, we have a single store per screen. This carries both benefits and drawbacks, both of which will be covered later in this document.
+
+### Important objects
+
+### <a name="store"/>Store
```

Comment:
I think I would also mention our `StoreProvider` which should be used to create a store. You did mention `ViewModel` below but probably good to point this out explicitly.